gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageReadChannel.java

```diff
@@ -1060,6 +1060,42 @@ protected InputStream openStream(long bytesToRead) throws IOException {
     }
   }
 
+  private void cacheFooterWithRetries(Storage.Objects.Get getObject, HttpResponse response)
```

Comment:
Wouldn't it be better to put retires inside `cacheFooter` while reading footer content from response's `contentStream`  instead of retrying the process?